Fix pdh-based go checks, when variable instances are added & removed

derekwbrown opened the pull request:
### What does this PR do?

Previously, available instances were only computed at `config` time; so if an instance was added (e.g. a drive plugged in), then it was missed by subsequent checks

### Motivation

customer issue

### Additional Notes

Also adds mocked version of PDH functions, and adds some unit tests.

derekwbrown committed:
Rewrite pdh base checks to allow for instances coming & going for
multi-instance counters

derekwbrown committed:
Add mocked-out version of PDH functions for unit tests

Add basic unit test for file_handles & iostats

achntrl commented on pkg/logs/pipeline/registry.json:
```diff
@@ -0,0 +1 @@
+{"Version":2,"Registry":{}}
```
Not sure you wanted to commit this file

derekwbrown committed:
Remove unnecessary file that got caught in commit

derekwbrown committed:
Add releasenotes

derekwbrown committed:
Add tests for winproc

Fix bug in mocks for single instance counter

Fix bug in returning when counter instance disappears

derekwbrown committed:
Fix tests to run on machine that don't have the added drive

hush-hush reviewed (commented):
Can't really review the PDH logic, left a few comments

derekwbrown commented on pkg/util/winutil/pdhutil/pdhcounter.go:
```diff
@@ -28,60 +31,35 @@ type CounterInstanceVerify func(string) bool
-	p.countermap = make(map[string]PDH_HCOUNTER)
-	var err error
+// Initialize initializes a counter set object
+func (p *PdhCounterSet) Initialize(className string) (err error) {
```
If I understand right, only `ndxlist` would be initialized on line 60... but I can change it...

derekwbrown committed:
Review feedback

hush-hush commented on pkg/util/winutil/pdhutil/pdhcounter.go:
```diff
@@ -91,71 +69,145 @@ func GetCounterSet(className string, counterName string, instanceName string, ve
-			winerror = PdhAddCounter(p.query, path, uintptr(0), &hc)
-			if ERROR_SUCCESS != winerror {
+	winerror := pfnPdhAddCounter(p.query, path, uintptr(0), &p.singleCounter)
+	if ERROR_SUCCESS != winerror {
```
OK !
